draft-ietf-quic-recovery.md

```diff
@@ -623,10 +623,21 @@ time.
 
 ### Sending Probe Packets
 
-When a PTO timer expires, the sender MUST send one ack-eliciting packet as a
-probe, unless there is nothing to send. A sender MAY send up to two
-ack-eliciting packets, to avoid an expensive consecutive PTO expiration due
-to a single packet loss.
+When a PTO timer expires, a sender MUST send at least one ack-eliciting packet
+as a probe, unless there is no data available to send.  An endpoint MAY send up
+to two ack-eliciting packets, to avoid an expensive consecutive PTO expiration
+due to a single packet loss.
+
+It is possible that the sender has no new or previously-sent data to send.  For
+example, this can happen when there is no new application data to send and all
+streams with data in flight are reset (see Section 13.2 of {{QUIC-TRANSPORT}}).
```

Comment:
As Nick Banks has repeatedly pointed out, when you reset a stream, you send a STREAM_RESET frame, which is ack-eliciting, so this example doesn't really hold up that well.  A MESSAGE/DATAGRAM frame is the easiest real-world example because it's ack-eliciting and unreliable, but it's not in the v1 draft :(

Reply:
Maybe the best example is if the PTO timer expires and the only thing left outstanding was a spurious retransmission. For instance, packet 1 sent a stream frame, then you thought it was lost, then you sent packet 2 but then an ACK came for packet 1. Then PTO fires. You can mark that packet as possibly lost, but it wouldn't cause anything to be queued for transmit.

Reply:
Because of the reset issue, I think another example, such as Nick's, would be best.

Reply:
Yeah, good point. Thanks for the alternate example, @nibanks.